azure/Guidelines.md

```diff
@@ -477,6 +477,210 @@ Overall, this is a very brittle design that leads to a poor developer experience
 
 ### Collections
 
+The standard REST pattern for accessing a collection of resources is with a _GET_ method on the collection URL.
+This operation is commonly called the "list operation" for the resource, since the primary content of the response
+is an array (list) of the resource type.  Query parameters on the list operation may be provided to control
+which resources are returned and in what order.
+
+Resource collections can often be arbitrarily large, increasing lookup time as well as the size of the responses being sent over the wire. Therefore, it is important that list operations implement pagination.
```

Comment:
We should probably distinguish between server driven pagination and client driven pagination.

Reply:
One of the points brought up in the first review was to consider simplifying our pagination guidance.  That's what I've done here.  You can think of this as "unified pagination".  The client controls the number of items in a single response with _maxpagesize_ rather than _top_.  Since the client has to handle server pagination anyway, there is no need for _top_.

Reply:
If we are going to choose between _top_ and _maxpagesize_, here's something to consider.  Currently there are only two instances of a _maxpagesize_ parameter in azure-rest-api-specs, but many (> 500) instances of _top_ (actually most of these are _$top_, but morally equivalent).

Reply:
I'd prefer `maxPageSize` if that's the semantic.  Every time we see `top` or `$top` on the client we have to have a discussion about exactly what that means.

We might also want to talk about not exposing server pagination parameters if you only need to expose `nextLink`.  Everyone gets that wrong.

Reply:
Decision: Remove maxPageSize in favor of top.